op-supervisor/supervisor/backend/l1access/l1_accessor.go

```diff
@@ -159,7 +162,24 @@ func (p *L1Accessor) onFinalized(ctx context.Context, ref eth.L1BlockRef) {
 }
 
 func (p *L1Accessor) onLatest(ctx context.Context, ref eth.L1BlockRef) {
+	// The block is the same one we already have
+	if ref.Number == p.tipHeight && ref.Hash == p.tipHash {
+		return
+	}
+
+	// The block is the next one; check if it is a reorg
+	if ref.Number == p.tipHeight+1 && ref.ParentHash != p.tipHash {
+		// Signal a safe block reorg
+		// return now because we don't want to update the tip info yet
+		p.emitter.Emit(superevents.InvalidateCrossSafeEvent{
+			// ChainID:   chainID,
+			// Candidate: ref,
+		})
+		return
```

Comment:
This shouldn't *just* invalidate cross safe, but local safe as well. If the L1 data has changed out from under us, then all databases should be reversed:
- Local Safe (because it represents what was in the L1, which is now different)
- Cross Safe (which is just Local Safe with extra rules)
- Unsafe/Event (which should have the same content as safe, but extends farther)

I need to go see how this Event is being handled, perhaps it is already correct but named confusingly.